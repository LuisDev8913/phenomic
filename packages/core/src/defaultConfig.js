// @flow

import normalizeBaseUrl from "./configuration/normalize-base-url.js";

const defaultConfig: PhenomicConfig = {
  baseUrl: normalizeBaseUrl("http://localhost"),
  path: process.cwd(),
  content: { content: ["**/*"] },
  outdir: "dist",
  port: 3333,
  bundleName: "phenomic.main",
  plugins: []
};
export default defaultConfig;
