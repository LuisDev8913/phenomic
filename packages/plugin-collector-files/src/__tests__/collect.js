// @flow

import createDB from "@phenomic/core/lib/db";
import defaultConfig from "@phenomic/core/lib/defaultConfig.js";

import collector from "..";

const db = createDB({});

const config = {
  ...defaultConfig,
  path: __dirname,
  content: {
    "": { root: "__fixtures__", globs: ["**/*"] },
  },
};

const p = collector(config, {});
it("should collect everything", async () => {
  if (p.collect) {
    await p.collect({
      db,
      transformers: [
        {
          name: "@phenomic/plugin-default-transform",
          supportedFileTypes: ["json"],
          transform({ contents }) {
            return JSON.parse(contents.toString());
          },
        },
      ],
    });
  }

  expect(await db.getList("__null__")).toMatchSnapshot();
});

it("should collect specific path", async () => {
  expect(await db.getList("news/2017")).toMatchSnapshot();
});

it("should collect metadata", async () => {
  expect(await db.getList("showcaseTags")).toMatchSnapshot();
});

it("should collect specific item", async () => {
  expect(
    await db.get("__null__", "news/2017/06/introducing-1.0.0-alpha"),
  ).toMatchSnapshot();
});
