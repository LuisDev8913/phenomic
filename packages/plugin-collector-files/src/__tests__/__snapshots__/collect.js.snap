// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect everything 1`] = `
Array [
  Object {
    "filename": "acrowithbrian.com.json",
    "id": "showcase/entry/acrowithbrian.com",
    "showcaseTags": Array [
      "event",
      "blog",
      "learning",
      "business",
    ],
    "source": "https://github.com/brianrc/acrowithbrian-site",
    "title": "Brian Cruikshank Acro",
    "url": "https://www.acrowithbrian.com/",
  },
  Object {
    "filename": "anthonydugois.com__polynomic.json",
    "id": "showcase/entry/anthonydugois.com__polynomic",
    "showcaseTags": Array [
      "docs",
      "open-source",
    ],
    "source": "https://github.com/anthonydugois/polynomic",
    "title": "Polynomic",
    "url": "http://anthonydugois.com/polynomic/",
  },
  Object {
    "bio": "bloodyowl short bio",
    "body": "bloodyowl long bio",
    "filename": "bloodyowl.json",
    "id": "authors/bloodyowl",
    "name": "bloodyowl",
  },
  Object {
    "filename": "ced.photo.json",
    "id": "showcase/entry/ced.photo",
    "showcaseTags": Array [
      "blog",
      "business",
    ],
    "source": "https://github.com/xuopled/ced.photo",
    "title": "CeD photo",
    "url": "https://www.ced.photo",
  },
  Object {
    "filename": "event.json",
    "id": "showcaseTags/event",
    "name": "Event",
  },
  Object {
    "bio": "moox short bio",
    "body": "moox long bio",
    "filename": "moox.json",
    "id": "authors/moox",
    "name": "moox",
  },
  Object {
    "filename": "open-source.json",
    "id": "showcaseTags/open-source",
    "name": "Open Source",
  },
  Object {
    "filename": "submit.json",
    "id": "showcase/submit",
    "title": "Submit a website/app made with Phenomic to the Showcase",
  },
  Object {
    "authors": Array [
      "bloodyowl",
    ],
    "date": "2017-06-02T00:00:00.000Z",
    "filename": "introducing-1.0.0-alpha.json",
    "id": "news/2017/06/introducing-1.0.0-alpha",
    "title": "Introducing Phenomic 1.0.0 first alpha",
  },
]
`;

exports[`should collect metadata 1`] = `
Array [
  Object {
    "filename": "event.json",
    "id": "event",
    "name": "Event",
  },
  Object {
    "filename": "open-source.json",
    "id": "open-source",
    "name": "Open Source",
  },
]
`;

exports[`should collect specific item 1`] = `
Object {
  "id": "news/2017/06/introducing-1.0.0-alpha",
  "value": Object {
    "authors": Array [
      Object {
        "bio": "bloodyowl short bio",
        "body": "bloodyowl long bio",
        "filename": "bloodyowl.json",
        "id": "bloodyowl",
        "name": "bloodyowl",
      },
    ],
    "body": Object {
      "c": Array [
        "Hi there. Alpha blah blah.",
      ],
      "t": "div",
    },
    "date": "2017-06-02T00:00:00.000Z",
    "filename": "introducing-1.0.0-alpha.json",
    "title": "Introducing Phenomic 1.0.0 first alpha",
  },
}
`;

exports[`should collect specific path 1`] = `
Array [
  Object {
    "authors": Array [
      "bloodyowl",
    ],
    "date": "2017-06-02T00:00:00.000Z",
    "filename": "introducing-1.0.0-alpha.json",
    "id": "06/introducing-1.0.0-alpha",
    "title": "Introducing Phenomic 1.0.0 first alpha",
  },
]
`;
